sdk/daml-script/test/src/main/scala/com/daml/lf/engine/script/test/IdeLedgerUpgradesIT.scala

```diff
@@ -0,0 +1,384 @@
+  final override protected lazy val timeMode = ScriptTimeMode.WallClock
+
+  final override protected lazy val devMode = true
+  final override protected val disableUpgradeValidation = true
+
+  // TODO(https://github.com/digital-asset/daml/issues/18457): split the test into one with contract
+  //  keys and one without, and revert to the default version. Here and below, in the loaded dars.
+  val languageVersion: LanguageVersion = LanguageVersion.v2_dev
+  override val majorLanguageVersion: LanguageMajorVersion = languageVersion.major
+
+  override protected lazy val darFiles = List()
+
+  lazy val damlScriptDar = requiredResource("daml-script/daml3/daml3-script-2.dev.dar")
+  lazy val upgradeTestLibDar: Path = rlocation(Paths.get("daml-script/test/upgrade-test-lib.dar"))
+
+  lazy val tempDir: Path = Files.createTempDirectory("upgrades-it")
+
+  val testFileDir: Path = rlocation(Paths.get("daml-script/test/daml/upgrades/"))
+  val testCases: Seq[TestCase] = getTestCases(testFileDir)
+
+  private def traverseSequential[A, B](elems: Seq[A])(f: A => Future[B]): Future[Seq[B]] =
+    elems.foldLeft(Future.successful(Seq.empty[B])) { case (comp, elem) =>
+      comp.flatMap { elems => f(elem).map(elems :+ _) }
+    }
+
+  // Maybe provide our own tracer that doesn't tag, it makes the logs very long
+  "Multi-participant Daml Script Upgrades" should {
+    testCases.foreach { testCase =>
+      testCase.name in {
+        for {
+          // Build dars
+          (testDarPath, deps) <- buildTestCaseDar(testCase)
+
+          // Connection
+          // clients <- scriptClients(provideAdminPorts = true)
+          // adminClients = ledgerPorts.map { portInfo =>
+          //   (
+          //     portInfo.ledgerPort.value,
+          //     TestingAdminLedgerClient.singleHost(
+          //       "localhost",
+          //       portInfo.adminPort.value,
+          //       None,
+          //       LedgerClientChannelConfiguration.InsecureDefaults,
+          //     ),
+          //   )
+          // }
+
+          // _ <- traverseSequential(adminClients) { case (ledgerPort, adminClient) =>
+          //   Future.traverse(deps) { dep =>
+          //     Thread.sleep(500)
+          //     println(
+          //       s"Uploading ${dep.versionedName} to participant on port ${ledgerPort}"
+          //     )
+          //     adminClient
+          //       .uploadDar(dep.path.toFile)
+          //       .map(_.left.map(msg => throw new Exception(msg)))
+          //   }
+          // }
+
+          // // Wait for upload
+          // _ <- RetryStrategy.constant(attempts = 20, waitTime = 1.seconds) { (_, _) =>
+          //   // assertDepsVetted(adminClients.head._2, deps)
+          //   ()
+          // }
+          // _ = println("All packages vetted on all participants")
+
+          // Run tests
+          testDar = CompiledDar.read(testDarPath, Runner.compilerConfig(LanguageMajorVersion.V2))
+          _ <- run(
+            clients,
```

Comment:
I need to figure out if I can call this with an IDE Ledger client (and if so, how to create one), or if I should just invoke `daml test` directly